src/mapping.ts

```diff
@@ -92,16 +126,19 @@ function deregisterPodLogic(id: string): void {
     oldAdmin.save();
   }
 
-  // Delete the Pod
-  store.remove('Pod', id);
-}
+  // Delete the PodUsers for this Pod
+  if (pod.users.length > 0) {
+    for (let i = 0; i < pod.users.length; i++) {
+      // Delete the PodUser
+      store.remove("PodUser", pod.users[i].toString());
+    }
+  }
```

Comment:
🎨 This logic (unrelated to transferBatch) was missing and causing an existing test to fail.

Reply:
good catch

Reply:
I previously had a bunch of logic to filter out the PodUser ID from the user's pod array, but reading [this](https://thegraph.com/docs/en/developing/creating-a-subgraph/#reverse-lookups) makes me think I don't need to (and shouldn't) since it's derived from other data.

In the case of `user.adminPods`, we had to do manual filtering because that's a string array that isn't derived.